### PR TITLE
Refactor code related to tutor dashboard

### PR DESCRIPTION
**Pull request by rpadovani**
### Checklist
- [x] I run `yarn run webpack:build:main`: the project builds without errors.
- [x] I run `yarn lint`: there project builds without code style warnings.
- [x] I removed unnecessary whitespace changes in all related files.
- [x] ~I updated the documentation and models.~
- [x] I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.
- [x] ~I added (end-to-end) test cases for the new functionality.~

### Motivation and Context
When I first implemented the tutor dashboards I left some TODOs, now I have addressed them

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/web/rest/TextSubmissionResource.java [hunk 1]
```diff
@@ -171,68 +172,51 @@ private void hideDetails(@RequestBody TextSubmission textSubmission) {
...
-     * @return the ResponseEntity with status 200 (OK) and the list of textSubmissions in body
-     */
-    @GetMapping(value = "/exercises/{exerciseId}/text-submissions-assessed-by-tutor")
-    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
-    @Transactional(readOnly = true)
-    public ResponseEntity<List<TextSubmission>> getAllTextSubmissionsByTutor(@PathVariable Long exerciseId) {
-        log.debug("REST request to get all TextSubmissions filtered by assessed by tutor");
-        Exercise exercise = exerciseService.findOneLoadParticipations(exerciseId);
-        User user = userService.getUserWithGroupsAndAuthorities();
+            Hibernate.initialize(textSubmission.getResult()); // eagerly load the association
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/web/rest/TextSubmissionResource.java]: these calls to Hibernate.initialize might be expensive if there are many participations (assume in EIST we have 2000 of them). Can we refactor this and instead eagerly load them together with the participations? I.e. change the logic of line 199 and use different database calls depending on the two parameters? Also there is too much logic involve here for a simple resource, so I suggest to move this into the corresponding service.

File: src/main/java/de/tum/in/www1/artemis/web/rest/TextSubmissionResource.java [hunk 2]
```diff
@@ -171,68 +172,51 @@ private void hideDetails(@RequestBody TextSubmission textSubmission) {
...
      */
     @GetMapping(value = "/exercises/{exerciseId}/text-submissions")
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
     @Transactional(readOnly = true)
     public ResponseEntity<List<TextSubmission>> getAllTextSubmissions(@PathVariable Long exerciseId,
-                                                                      @RequestParam(defaultValue = "false") boolean submittedOnly) {
+                                                                      @RequestParam(defaultValue = "false") boolean submittedOnly,
+                                                                      @RequestParam(defaultValue = "false") boolean assessedByTutor) {
         log.debug("REST request to get all TextSubmissions");
         Exercise exercise = exerciseService.findOneLoadParticipations(exerciseId);
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/web/rest/TextSubmissionResource.java]: You don't need to load the exercise with all participations here as you load the participations in a different call later on. So please only use findOne(exerciseId) here

File: src/main/webapp/app/text-assessment/text-assessment.component.ts [hunk 3]
```diff
@@ -50,17 +50,34 @@ export class TextAssessmentComponent implements OnInit, OnDestroy {
...
+        if (submissionValue === 'new') {
+            this.assessmentsService.getParticipationForSubmissionWithoutAssessment(exerciseId).subscribe(participation => {
+                this.participation = participation;
+                this.result = new Result();
+                this.submission = <TextSubmission>this.participation.submissions[0];
+                this.exercise = <TextExercise>this.participation.exercise;
+                this.assessments = [];
+                this.busy = false;
+
+                // Update the url with the new id, without reloading the page, to make the history consistent
```
Comment (jpbernius) [inline src/main/webapp/app/text-assessment/text-assessment.component.ts]: :+1:

File: src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java [hunk 4]
```diff
@@ -183,6 +186,29 @@ public ParticipationResource(ParticipationService participationService,
...
+     * GET /exercise/{exerciseId}/participation-without-assessment
+     *
+     * Given an exerciseId, retrieve a participation where the latest submission has no assessment, or returns 404
+     *
+     * @param exerciseId the id of the exercise of which we want a submission
+     * @return a student participation
+     */
+    @GetMapping(value = "/exercise/{exerciseId}/participation-without-assessment")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Participation> getParticipationForExerciseWithoutAssessment(@PathVariable Long exerciseId) {
```
[the remark this answers is not in the document]
Comment (rpadovani) [reply src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java]: Right, done

File: src/main/java/de/tum/in/www1/artemis/web/rest/TextSubmissionResource.java (hunk 2 above)
Comment (rpadovani) [reply src/main/java/de/tum/in/www1/artemis/web/rest/TextSubmissionResource.java]: Done

File: src/main/java/de/tum/in/www1/artemis/web/rest/TextSubmissionResource.java (hunk 1 above)
Comment (rpadovani) [reply src/main/java/de/tum/in/www1/artemis/web/rest/TextSubmissionResource.java]: I moved to a service and improved the query

**Pull request merged**